chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1337,6 +1405,30 @@ pub(crate) async fn blocks_with_triggers(
     let trigger_futs: FuturesUnordered<BoxFuture<Result<Vec<EthereumTrigger>, anyhow::Error>>> =
         FuturesUnordered::new();
 
+    // This is for `start` triggers which can be initialization handlers which needs to be run
+    // before all other triggers
+    if filter.block.trigger_every_block {
+        let block_future = eth
+            .block_range_to_ptrs(logger.clone(), from, to)
```

Comment:
When `trigger_every_block` is set, this is now calling `block_range_to_ptrs` twice, once for Start triggers and then again for End triggers, which is very inefficient. Lets call it only once, and use the result both for the Start and End triggers.